### PR TITLE
[ticket/14466] Add an event to cron.php

### PR DESCRIPTION
**Pull request by lavigor**
Should replace [PHPBB3-14012](https://tracker.phpbb.com/browse/PHPBB3-14012).

[PHPBB3-14466](https://tracker.phpbb.com/browse/PHPBB3-14466)

### CODE REVIEW

**Comment by Nicofuma** [comment]
You want to use this event to change which task should be run? why?

**Comment by lavigor** [comment]
!set Event

**Comment by lavigor** [comment]
@Nicofuma No, I want to write its name to the database so that if the task isn't finished, the user will know what to check.

**Comment by Nicofuma** [comment]
ok

**Comment by lavigor** [comment]
@Nicofuma [here](https://github.com/phpbb/phpbb/pull/4173/files#diff-ba8bda93858e9f8b9f5cc68ddca1ad5cR62) I wrote `\phpbb\cron\task\task` because of [this line](https://github.com/phpbb/phpbb/blob/3.1.x/phpBB/phpbb/cron/manager.php#L113).
However, I noticed that all tasks in the manager are returned wrapped because of [this line](https://github.com/phpbb/phpbb/blob/3.1.x/phpBB/phpbb/cron/manager.php#L61).

Then I think that `return` statement should be changed to `\phpbb\cron\task\wrapper` in the manager and this PR. Should I fix it here or create a separate PR?

**Comment by marc1706** [comment]
Please create another ticket for other incorrect docblocks.

File: phpBB/cron.php [hunk 1]
```diff
@@ -55,6 +55,18 @@ function output_image()
 	$task = $cron->find_task($cron_type);
 	if ($task)
 	{
+		/**
+		 * This event enables you to catch the task before it runs
+		 *
+		 * @event core.cron_run_before
+		 * @var	\phpbb\cron\task\task	task	Current Cron task
```
Comment (marc1706) [inline phpBB/cron.php]: Should be `\phpbb\cron\task\wrapper` as you pointed out.

**Comment by lavigor** [comment]
!unset WIP

**Comment by lavigor** [comment]
@marc1706 are events possible in CLI?
Or should they be avoided there?

I ask because Cron can be run through CLI too and this event won't be executed in that case.

**Comment by Nicofuma** [comment]
Yes you can

**Pull request merged**